packages/cozy-client/src/authentication/mobile.js

```diff
@@ -25,8 +25,8 @@ export const authenticateWithCordova = url => {
       }
 
       const onLoadStart = ({ url }) => {
-        const accessCode = /\?code=(.+)$/.test(url)
-        const state = /\?state=(.+)$/.test(url)
+        const accessCode = /\bcode=/.test(url)
+        const state = /\bstate=/.test(url)
```

Comment:
It's been several times that we have trouble parsing URLs with regexes. I think it's better if we use a dedicated url parser. It  will also be more readable.

Reply:
:+1: it would be a lot better to parse URLs with a dedicated URL parser. It's just that this code is in cordova and I don't know what APIs are available here, or how to include an external module.

Reply:
OK I will open a dedicated issue for this.